src/odemis/acq/__init__.py

```diff
@@ -396,6 +398,14 @@ def _adjust_metadata(self, raw_data):
                 if model.MD_DESCRIPTION not in d.metadata:
                     d.metadata[model.MD_DESCRIPTION] = s.name.value
 
+        # Add extra settings to metadata
+        for s, data in raw_data.items():
+            for d in data:
+                if hasattr(s, '_md_manager') and s._md_manager:
+                    d.metadata[model.MD_EXTRA_SETTINGS] = s._md_manager.all_metadata
+                else:
+                    logging.warning("Stream %s has no metadata manager, not saving extra metadata.", s)
```

Comment:
This is done only after the acquisition of all the streams. It should be done just after the stream acquisition, and *before* the acquisition of the next stream.